### PR TITLE
Feature/fed refactor

### PR DESCRIPTION
**Pull request by CSergienko**
@chrislombaard @phalamag Please have a look and let me know if I can merge to develop.

### CODE REVIEW

File: nurseconnect/templates/core/tags/bannerpages.html [hunk 1]
```diff
@@ -8,7 +8,7 @@
                     <header class="Paper-header">
                         <h1 class="Paper-headline">{{ bannerpage.title }}</h1>
                         {% if bannerpage.banner_link_page %}
-                            <a href="{% pageurl bannerpage.banner_link_page %}">{% image bannerpage.banner max-500x150 class="Article-image Article-image--prominent" %}</a>
+                            <a href="{% pageurl bannerpage.banner_link_page %}">{% image bannerpage.banner max-500x150 class="Article-image Article-image--prominent" height="auto" %}</a>
```
Comment (CSergienko) [inline nurseconnect/templates/core/tags/bannerpages.html]: I think this snuck in from develop. height="auto" shouldn't be necessary now.

File: nurseconnect/forms.py [hunk 2]
```diff
@@ -187,13 +187,11 @@ def clean_username(self):
...
                 self.cleaned_data["username"] = raw_username
 
-        if self.user.username != self.cleaned_data["username"]:
-            if User.objects.filter(
-                username__iexact=self.cleaned_data["username"]
-            ).exists():
-                self._errors["username"] = self.error_class(
-                    ["Username already exists."]
-                )
+        # if not self.request.user.username == self.cleaned_data["username"]:
```
Comment (chrislombaard) [inline nurseconnect/forms.py]: Why did you add this? Line 190 on Develop does not reflect this.

File: nurseconnect/templates/core/tags/bannerpages.html (hunk 1 above)
Comment (chrislombaard) [reply nurseconnect/templates/core/tags/bannerpages.html]: Cool so that can be removed yes?

File: nurseconnect/forms.py (hunk 2 above)
Comment (chrislombaard) [reply nurseconnect/forms.py]:
```
def clean_username(self):
        username = self.cleaned_data["username"]
        if username:
            raw_username = username.raw_input
            if raw_username[0] == "0":
                self.cleaned_data["username"] = \
                    INT_PREFIX + raw_username[1:len(username)]
            else:
                self.cleaned_data["username"] = raw_username

        if self.user.username != self.cleaned_data["username"]:
            if User.objects.filter(
                username__iexact=self.cleaned_data["username"]
            ).exists():
                self._errors["username"] = self.error_class(
                    ["Username already exists."]
                )

        return self.cleaned_data["username"]
```

Comment (chrislombaard) [reply nurseconnect/forms.py]: Great formatting github. On point.

Comment (CSergienko) [reply nurseconnect/forms.py]: I didn't add anything. Must have been a merge hiccup.

File: nurseconnect/templates/core/tags/bannerpages.html (hunk 1 above)
Comment (CSergienko) [reply nurseconnect/templates/core/tags/bannerpages.html]: Yep

Comment (chrislombaard) [reply nurseconnect/templates/core/tags/bannerpages.html]: Cool, please remove then

**Comment by chrislombaard** [comment]
+1
